cdhweb/static_src/global/rich-text.scss

```diff
@@ -1,56 +1,95 @@
-  // typically in a hero banner rather than coming from a WYSIWYG with other text.
-  :is(h2, h3, h4, h5, h6) {
-    margin-top: var(--space-between-sections);
-  }
-  // ...unless a heading appears immediately after another heading, in which case we need to reduce the spacing.
-  h2 + h3,
-  h3 + h4,
-  h4 + h5,
-  h5 + h6 {
-    margin-top: var(--space-between-els, 1em);
-  }
-
-  // Remove any orphan vertical space at the start/end of the rich-text block.
-  // NOTE: this assumes items are direct children of .rich-text container.
-  > * {
-    &:first-child {
-      margin-top: 0;
-    }
-    &:last-child {
-      margin-bottom: 0;
-    }
-  }
-
-  ul,
-  ol {
-    // Undo padding:0 from the css reset, which kills list indentation.
-    padding: revert;
-  }
-
-  // Collapsing margins should ensure we don't get extra space between this and preceding element.
-  // i.e. no `:not(:first...)` required.
-  li {
-    margin-top: 0.5em;
+// By default, every prose element has a bottom margin.
+// `li` is excluded because list item spacing is handled later on.
+// Same with headings because they have their own, more specific spacing values.
+// Ignore header and footer, they have their own styles.
+:where(main) {
+  --space-default: 1rem;
+
+  --space-l2-lg: #{px2rem(56)};
+  --space-l3-lg: #{px2rem(40)};
+  --space-l4-lg: #{px2rem(24)};
+  --space-l5-lg: #{px2rem(40)};
+  --space-l6-lg: #{px2rem(40)};
+
+  --space-l1-sm: calc(var(--space-l1-lg) / 4);
+  --space-l2-sm: calc(var(--space-l2-lg) / 4);
+  --space-l3-sm: calc(var(--space-l3-lg) / 4);
+  --space-l4-sm: calc(var(--space-l4-lg) / 4);
+  --space-l5-sm: calc(var(--space-l5-lg) / 4);
+  --space-l6-sm: calc(var(--space-l6-lg) / 4);
+
+  --space-list-items: #{px2rem(8)};
+
+  @include md {
+    --space-l2-lg: #{px2rem(104)};
+  }
+
+  :is(p, ul, ol) {
+    margin-bottom: var(--space-default);
+  }
+  // Headings have a little more bottom spacing than other prose elements
+  h2 {
+    margin-bottom: var(--space-l2-sm);
+  }
+  h3 {
+    margin-bottom: var(--space-l3-sm);
+  }
+  h4 {
+    margin-bottom: var(--space-l4-sm);
+  }
+  h5 {
+    margin-bottom: var(--space-l5-sm);
+  }
+  h6 {
+    margin-bottom: var(--space-l6-sm);
+  }
+
+  *:has(+ h2) {
+    margin-bottom: var(--space-l2-lg);
+  }
+  *:has(+ h3) {
+    margin-bottom: var(--space-l3-lg);
+  }
+  *:has(+ h4) {
+    margin-bottom: var(--space-l4-lg);
+  }
+  *:has(+ h5) {
+    margin-bottom: var(--space-l5-lg);
+  }
+  *:has(+ h6) {
+    margin-bottom: var(--space-l6-lg);
+  }
+
+  // Headings followed immediately by a heading of a level down
+  // should have only a small gap below.
+  :where(h2):has(+ h3) {
+    margin-bottom: var(--space-l2-sm);
+  }
+  :where(h3):has(+ h4) {
+    margin-bottom: var(--space-l3-sm);
+  }
+  :where(h4):has(+ h5) {
+    margin-bottom: var(--space-l4-sm);
+  }
+  :where(h5):has(+ h6) {
+    margin-bottom: var(--space-l5-sm);
+  }
+
+  // List item spacing.
+  :where(li):has(+ li) {
```

Comment:
a couple of vague thoughts about this, I'll present them in a list so it's on-theme:
1. This feels like one of the more potentially leaky parts of the "no more `.rich-text`" system, just because we use lists in a bunch of places where we don't display them visually as lists, or display them horizontally etc. I still think it's pretty safe because we basically always have more specific rules/classes on the list and list items in those scenarios, but just wanted to flag that this felt slightly _less_ secure than the other elements
2. as list items are always nicely wrapped in a list container, we could use `flex`, `column`, `gap` on `ul, ol` to cover both spacing for the items and spacing for nested lists, without needed to check for `+ li`. It doesn't totally fit with your 'just margin-bottom' approach, but you're breaking that rule on line 86 anyway

Reply:
1. Yes, that is indeed the trade-off. So far it feels a little nicer in that there's slightly fewer instances of having-to-remove-margins than there were rich-text wrappers. 
2. Interesting idea. We'd still need a top-margin for nested lists though, as the nested `ul` is inside an `li`, therefore no gap is produced. See:
![Uploading Screenshot 2024-05-20 at 4.11.27 PM.png…]()

Reply:
ah, yeah , you're right

Reply:
I think I'll skip the `gap` thing for now. If there were a way to make `gap` work for the nested lists too I'd be more interested, but as it is we need a margin anyway.